From: Ben Hutchings <bhutchings@solarflare.com>
Date: Wed, 19 Sep 2012 00:31:18 +0100
Subject: sfc: Do not initialise buffer in efx_alloc_special_buffer()
Patch-mainline: v3.8-rc1
Git-commit: c3771a35bebeea16c4fd94d1116010cc02b1f75e
References: bnc#786035 FATE#314299

Currently we initialise the newly allocated buffer to all-1s, which is
important for event queues but not for descriptor queues.  And since
we also do that in efx_nic_init_eventq(), it is completely pointless
to do it here.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/sfc/nic.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/net/ethernet/sfc/nic.c
+++ b/drivers/net/ethernet/sfc/nic.c
@@ -263,9 +263,6 @@ static int efx_alloc_special_buffer(stru
 	buffer->entries = len / EFX_BUF_SIZE;
 	BUG_ON(buffer->dma_addr & (EFX_BUF_SIZE - 1));
 
-	/* All zeros is a potentially valid event so memset to 0xff */
-	memset(buffer->addr, 0xff, len);
-
 	/* Select new buffer ID */
 	buffer->index = efx->next_buffer_table;
 	efx->next_buffer_table += buffer->entries;
